python/ray/serve/controller.py

```diff
@@ -129,8 +136,56 @@ def get_http_proxies(self) -> Dict[NodeId, ActorHandle]:
         """Returns a dictionary of node ID to http_proxy actor handles."""
         return self.http_state.get_http_proxy_handles()
 
+    async def autoscale(self) -> None:
```

Comment:
actually there's no reason why this needs to be async and it's easier to reason about it if it isn't -- can we make both this and `self._deploy` synchronous instead?

Reply:
`self._deploy` contains an `async with self.write_lock`, but you're saying if we make it synchronous, we don't even need the lock there--that makes sense.